### ISSUE TITLE
[🐞] High Total Blocking Time (TBT) using Google Analytics and partytown

### ISSUE THREAD

**Issue by benmccann** [QwikDev/partytown#513]
### Describe the bug

![Screenshot from 2023-12-26 09-05-03](https://github.com/BuilderIO/partytown/assets/322311/19363b11-ccb6-4f89-b175-90e0f93af493)

![Screenshot from 2023-12-15 22-33-37](https://github.com/BuilderIO/partytown/assets/322311/d9c4740c-a6c2-41f5-9bc8-3aa92d4cf604)

I don't know if this is purely unavoidable. However, it surprised me as I thought Partytown was supposed to move all computations off the main thread.

My page speed score is a couple of points better without Partytown + GA. However, Partytown + GA is still several points better than GA alone.

### Reproduction

https://partytown.builder.io/google-tag-manager

### Steps to reproduce

Setup Google Analytics with Partytown. E.g. I followed https://partytown.builder.io/sveltekit

 

Then use https://pagespeed.web.dev/ (look under "Avoid long main-thread tasks") or https://webpagetest.org/ to see that Partytown is significantly blocking the main thread.

### Browser Info

Chrome

**Comment by benmccann** [comment]
Looks like Lighthouse has gotten stricter on its scoring. That amount of blocking time is now taking 7 points of the Lighthouse score

![Screenshot from 2024-01-09 08-09-52](https://github.com/BuilderIO/partytown/assets/322311/2e4b5e69-7c97-4b08-aec7-83137e5d2336)

**Comment by gioboa** [comment]
Can you try with the new release pls? Thanks

**Comment by benmccann** [comment]
Thank you for taking a look. Unfortunately no improvement

![Screenshot from 2024-01-09 14-13-37](https://github.com/BuilderIO/partytown/assets/322311/d142bfe2-28a6-4acb-bbbf-f4b6fe763b96)

**Comment by gioboa** [comment]
Here is my result. [This](https://c3.ventures/) is your website, right?
<img width="825" alt="Screenshot 2024-01-09 at 23 36 36" src="https://github.com/BuilderIO/partytown/assets/35845425/a0deb78a-c654-4fa7-b7e3-98a01391c291">

**Comment by benmccann** [comment]
Yes, that's my site that I've been testing with. The overall score is somewhat variable with some runs doing better than others. Obviously the site scores well overall, but I can't help but want to get to 100 :grin:  Probably the most useful piece to look at rather than the overall score is the "Avoid long main-thread tasks" section and see whether Partytown is listed there as contributing to Total Blocking Time. My expectation was that Partytown shouldn't block, but I have limited insight into the internals, so perhaps there's some reason it's not possible for Partytown to operate in a non-blocking way.

**Comment by stefan9541** [comment]
have the same issue, in my case overall TBT by partytown-sandbox-sw.htm same as I had with Google Analytics and google-tag-manager scripts together)

**Comment by SkyZeroZx** [comment]
I had the same issue with TBT when integrated G4 with Angular & Partytown 

![image](https://github.com/user-attachments/assets/d581ba30-d41b-4991-a798-092213c6db3d)

**Comment by gioboa** [comment]
Fixed in the next release: large non-blocking DOM operation batches now yield every ~40ms, so Partytown's main-thread work stays under the 50ms long-task threshold. Please retest and reopen if your TBT numbers don't improve.
